### PR TITLE
Remove immutable property and support for query param

### PR DESCRIPTION
**Pull request by marcianos**
# Description of change
Removed the immutable property from the DID Document type and related methods.
Add support for query parameter diff=true|false.

## Links to any relevant issues
fixes issue #141 

## Type of change
- [X] Enhancement (a non-breaking change which adds functionality)
- [X] Documentation Fix

## How the change has been tested
Customization and addition of new tests in diff_chain example.
Test with no query parameter, test with query parameter diff=true and diff=false.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-iota/src/client/client.rs [hunk 1]
```diff
@@ -160,16 +160,22 @@ impl Client {
 
     let auth: AuthChain = AuthChain::try_from_messages(did, &messages)?;
 
-    let diff: DiffChain = if auth.current().immutable() {
-      DiffChain::new()
-    } else {
+    // Check if there is any query given and return
+    let diff_option = match did.query() {
```
Comment (l1h3r) [inline identity-iota/src/client/client.rs:164]: You can use [`query_pairs`](https://docs.rs/did_url/0.1.0/did_url/struct.DID.html#method.query_pairs) for this - it's pretty explicit but it handles encoding automatically

```rust
let skip_diff: bool = did.query_pairs().any(|(key, _)| key == "diff" && value == "false").unwrap_or_default();
```

File: examples/diff_chain.rs [hunk 2]
```diff
@@ -179,5 +179,54 @@ async fn main() -> Result<()> {
...
+  println!("Chain (R) {:#}", remote);
+  println!();
+
+  let a: &Document = chain.current();
+  let b: &Document = remote.current();
+
+  // The current document in the resolved chain should not be identical to
+  // the current document in our local chain because you read an outdated
+  // version of the document
+  assert_ne!(a, b);
```
Comment (l1h3r) [inline examples/diff_chain.rs:229]: nice! always forget about `assert_ne!`

File: identity-iota/src/client/client.rs (hunk 1 above)
Comment (marcianos) [reply identity-iota/src/client/client.rs:164]: Ah thank you! I didn't see query_pairs. But is has to be:
`let skip_diff: bool = did.query_pairs().any(|(key, value)| key == "diff" && value == "false");`
Is that right?

Comment (l1h3r) [reply identity-iota/src/client/client.rs]: Yeah, that looks right 👍

**Review by JelleMillenaar — APPROVED** [review]
Approved based on preview review from @l1h3r's feedback being implemented.

**Pull request merged**